src/sinks/util/http.rs

```diff
@@ -90,8 +90,14 @@ where
     type SinkError = ();
 
     fn start_send(&mut self, item: Self::SinkItem) -> StartSend<Self::SinkItem, Self::SinkError> {
+        if self.slot.is_some() {
+            self.poll_complete()?;
+            return Ok(AsyncSink::NotReady(item));
+        }
```

Comment:
I had to read this three times (and the struct definition) before I clued in what was going on here. Some commentary about the `slot` handling here and below would probably be appropriate.

Reply:
I added some more docs to the struct, but the `slot` item on the struct has a comment and links to the `with_flat_map` contaminator as well where this trick is handled.